### ISSUE TITLE
Analyze Images not using CPU or making progress

### ISSUE THREAD

**Issue by dana-mcc** [CellProfiler/CellProfiler#4863]
Hi, I'm seeing really slow performance using CellProfiler 4.2.6 on Linux Ubuntu 20.04. Specifically, using a pipeline with just 6 image sets, with each image set containing 13 channels being about 150MB total (900MB all sets). The current version of the pipeline is only using 4 images from each set. The pipeline had processed 0 images after 12 hours and was projected to take over 36 hours. When I checked the system monitor, no CPU was being used by CellProfiler. Our machine has 128G of RAM and 20-core CPU. I am able to run all of these modules in Test Mode, just not when I try to Analyze Images.

I've attached images of the modules I am using and the progress bar. Can you please advise on how we can speed this up?
![modules](https://github.com/CellProfiler/CellProfiler/assets/114681857/37f44e55-5358-4bd3-9dac-fda9b597f205)
![progress_bar](https://github.com/CellProfiler/CellProfiler/assets/114681857/ce8488e0-f2c4-4cc5-807b-54e57c113c86)

CellProfiler was installed with these instructions: https://github.com/CellProfiler/CellProfiler/wiki/Ubuntu-20.04

Below is the output of pip list:
Package                   Version
------------------------- --------------------
absl-py                   1.4.0
aiofiles                  22.1.0
aiohttp                   3.8.5
aiosignal                 1.3.1
aiosqlite                 0.19.0
anndata                   0.8.0
annotated-types           0.6.0
anyio                     3.7.1
apturl                    0.5.2
argon2-cffi               21.3.0
argon2-cffi-bindings      21.2.0
arrow                     1.2.3
asttokens                 2.0.8
async-timeout             4.0.2
attrs                     23.1.0
Babel                     2.12.1
backcall                  0.2.0
bash_kernel               0.9.1
bcrypt                    3.1.7
beautifulsoup4            4.11.1
bio                       1.5.9
biopython                 1.81
biothings-client          0.3.0
bleach                    5.0.1
blinker                   1.4
boto3                     1.34.34
botocore                  1.34.34
Brlapi                    0.7.0
build                     0.10.0
CacheControl              0.12.14
cached-property           1.5.2
cachy                     0.3.0
cdo                       1.6.0
CellProfiler              4.2.6
cellprofiler-core         4.2.6
centrosome                1.2.2
certifi                   2019.11.28
cffi                      1.15.1
cftime                    1.6.2
chardet                   3.0.4
charset-normalizer        3.2.0
chex                      0.1.7
cleo                      0.8.1
Click                     7.0
clikit                    0.6.2
cmake                     3.27.0
colorama                  0.4.3
coloredlogs               15.0.1
command-not-found         0.3
constants                 0.6.0
contextlib2               21.6.0
contourpy                 1.0.6
crashtest                 0.3.1
cryptography              41.0.3
cupshelpers               1.0
cycler                    0.11.0
daal                      2023.2.1
daal4py                   2023.2.1
dbus-python               1.2.16
debugpy                   1.6.3
decorator                 5.1.1
defer                     1.0.6
defusedxml                0.7.1
Deprecated                1.2.14
deprecation               2.1.0
distlib                   0.3.7
distro                    1.4.0
distro-info               0.23ubuntu1
dm-tree                   0.1.8
dnspython                 2.2.1
docrep                    0.3.2
docstring-parser          0.15
docutils                  0.15.2
dulwich                   0.21.5
duplicity                 0.8.12.0
emoji                     2.7.0
ensembl-rest              0.3.3
entrypoints               0.3
esm-tools                 6.18.3
et-xmlfile                1.1.0
etils                     1.3.0
exceptiongroup            1.1.2
executing                 1.1.0
f90nml                    1.4.3
fair-esm                  2.0.0
faiss-gpu                 1.7.2
fasteners                 0.14.1
fastjsonschema            2.16.2
filelock                  3.12.2
flax                      0.7.1
fonttools                 4.38.0
fqdn                      1.5.1
frozenlist                1.4.0
fsspec                    2023.6.0
future                    0.18.2
gfw-creator               0.2.2
gitdb                     4.0.10
GitPython                 3.1.32
gprofiler-official        1.0.0
greenlet                  2.0.2
h5py                      3.6.0
html5lib                  1.0.1
httplib2                  0.14.0
humanfriendly             10.0
hyperopt                  0.1.2
idna                      2.8
igraph                    0.10.6
imageio                   2.33.1
importlib-metadata        4.12.0
importlib-resources       5.9.0
inflect                   6.2.0
installer                 0.7.0
intervaltree              3.1.0
ipdb                      0.13.13
ipykernel                 6.16.0
ipython                   8.5.0
ipython-genutils          0.2.0
ipywidgets                6.0.0
isoduration               20.11.0
jaraco.classes            3.3.0
jax                       0.4.13
jaxlib                    0.4.13
jedi                      0.18.1
jeepney                   0.8.0
Jinja2                    3.1.2
jmespath                  1.0.1
joblib                    1.2.0
json5                     0.9.14
jsonpointer               2.4
jsonschema                4.18.6
jsonschema-specifications 2023.7.1
jupyter                   1.0.0
jupyter_client            8.3.0
jupyter-console           6.6.3
jupyter_core              5.3.1
jupyter-events            0.7.0
jupyter-resource-usage    1.0.0
jupyter_server            2.7.0
jupyter_server_fileid     0.9.0
jupyter_server_terminals  0.4.4
jupyter_server_ydoc       0.8.0
jupyter-ydoc              0.2.5
jupyterlab                3.6.5
jupyterlab-executor       2021.1.6
jupyterlab-pygments       0.2.2
jupyterlab_server         2.24.0
keyring                   21.8.0
kiwisolver                1.4.4
language-selector         0.1
launchpadlib              1.10.13
lazr.restfulclient        0.14.2
lazr.uri                  1.0.3
leidenalg                 0.10.1
lightning-utilities       0.9.0
lit                       16.0.6
llvmlite                  0.39.1
lockfile                  0.12.2
loguru                    0.7.0
louis                     3.12.0
louvain                   0.8.1
lxml                      4.9.1
macaroonbakery            1.3.1
mahotas                   1.4.13
Mako                      1.1.0
markdown-it-py            3.0.0
MarkupSafe                2.1.1
matplotlib                3.6.2
matplotlib-inline         0.1.6
mc-bin-client             1.0.1
mdurl                     0.1.2
mistune                   2.0.4
ml-collections            0.1.1
ml-dtypes                 0.2.0
monotonic                 1.5
more-itertools            4.2.0
mpmath                    1.3.0
msgpack                   1.0.5
mudata                    0.2.3
multidict                 6.0.4
multipledispatch          1.0.0
mygene                    3.2.2
mypy-extensions           1.0.0
mysqlclient               1.4.6
natsort                   8.2.0
nbclassic                 1.0.0
nbclient                  0.6.8
nbconvert                 7.0.0
nbformat                  5.6.1
nest-asyncio              1.5.5
netCDF4                   1.6.4
netifaces                 0.10.4
networkx                  3.0
notebook                  6.0.3
notebook_shim             0.2.3
numba                     0.56.4
numpy                     1.23.3
numpyro                   0.12.1
nvidia-cublas-cu11        11.10.3.66
nvidia-cuda-cupti-cu11    11.7.101
nvidia-cuda-nvrtc-cu11    11.7.99
nvidia-cuda-runtime-cu11  11.7.99
nvidia-cudnn-cu11         8.5.0.96
nvidia-cufft-cu11         10.9.0.58
nvidia-curand-cu11        10.2.10.91
nvidia-cusolver-cu11      11.4.0.1
nvidia-cusparse-cu11      11.7.4.91
nvidia-nccl-cu11          2.14.3
nvidia-nvtx-cu11          11.7.91
oauthlib                  3.1.0
olefile                   0.46
openpyxl                  3.1.2
opt-einsum                3.3.0
optax                     0.1.7
orbax-checkpoint          0.2.3
overrides                 7.3.1
packaging                 20.9
pandas                    1.5.0
pandocfilters             1.5.0
paramiko                  2.6.0
parso                     0.8.3
pastel                    0.2.1
pathlib                   1.0.1
patsy                     0.5.3
pexpect                   4.8.0
pickleshare               0.7.5
pillow                    10.2.0
pip                       23.3.2
pkginfo                   1.9.6
pkgutil_resolve_name      1.3.10
platformdirs              3.10.0
plotly                    5.15.0
poetry                    1.1.7
poetry-core               1.0.8
poetry-plugin-export      1.4.0
pooch                     1.7.0
prokaryote                2.4.4
prometheus-client         0.14.1
prompt-toolkit            3.0.31
protobuf                  3.6.1
psutil                    5.9.5
ptyprocess                0.7.0
pure-eval                 0.2.2
pycairo                   1.16.2
pycparser                 2.21
pycups                    1.9.73
pydantic                  2.6.0
pydantic_core             2.16.1
PyGithub                  1.59.0
Pygments                  2.13.0
PyGObject                 3.36.0
PyJWT                     2.8.0
pylev                     1.4.0
pymacaroons               0.13.0
pymongo                   4.3.3
PyNaCl                    1.5.0
pynndescent               0.5.8
pyparsing                 3.0.9
pyproject_hooks           1.0.0
pyRFC3339                 1.1
pyro-api                  0.1.2
pyro-ppl                  1.8.6
pyrsistent                0.18.1
python-apt                2.0.0+ubuntu0.20.4.8
python-bioformats         4.0.7
python-dateutil           2.8.2
python-debian             0.1.36ubuntu1
python-javabridge         4.0.3
python-json-logger        2.0.7
pytorch-lightning         1.9.5
pytz                      2022.2.1
PyWavelets                1.4.1
pyxdg                     0.26
PyYAML                    6.0.1
pyzmq                     22.3.0
qtconsole                 5.4.3
QtPy                      2.3.1
questionary               1.10.0
rapidfuzz                 2.15.1
record-keeper             0.9.32
referencing               0.30.0
reportlab                 3.5.34
requests                  2.31.0
requests-toolbelt         0.9.1
requests-unixsocket       0.2.0
rfc3339-validator         0.1.4
rfc3986-validator         0.1.1
rich                      13.5.2
rpds-py                   0.9.2
s3transfer                0.10.0
scanpy                    1.9.1
scikit-image              0.18.3
scikit-learn              0.24.2
scikit-learn-intelex      2023.2.1
scikit-misc               0.1.4
scipy                     1.9.0
screen-resolution-extra   0.0.0
scvi                      0.6.8
scvi-tools                0.20.3
seaborn                   0.12.2
SecretStorage             3.3.3
semver                    3.0.1
Send2Trash                1.8.0
sentry-sdk                0.18.0
session-info              1.0.0
setuptools                67.7.2
shellingham               1.5.0.post1
simplejson                3.16.0
six                       1.14.0
smmap                     5.0.0
sniffio                   1.3.0
sortedcontainers          2.4.0
soupsieve                 2.3.2.post1
SQLAlchemy                2.0.19
stack-data                0.5.1
statsmodels               0.13.5
stdlib-list               0.8.0
sympy                     1.12
systemd-python            234
tabulate                  0.9.0
tap                       0.2
tbb                       2021.10.0
tenacity                  8.2.2
tensorstore               0.1.40
terminado                 0.15.0
testpath                  0.4.4
texttable                 1.6.7
threadpoolctl             3.1.0
tifffile                  2022.4.8
tinycss2                  1.1.1
toml                      0.10.2
tomli                     2.0.1
tomlkit                   0.12.1
toolz                     0.12.0
torch                     1.12.1
torchaudio                0.12.1
torchdata                 0.6.1
torchmetrics              1.0.1
torchtext                 0.15.2
torchvision               0.13.1
tornado                   6.2
tqdm                      4.64.1
traitlets                 5.9.0
triton                    2.0.0
trove-classifiers         2023.8.7
typed-argument-parser     1.8.1
typing_extensions         4.9.0
typing-inspect            0.9.0
ubuntu-advantage-tools    27.10
ubuntu-drivers-common     0.0.0
ufw                       0.36
umap-learn                0.5.3
unattended-upgrades       0.1
uri-template              1.3.0
urllib3                   1.26.16
usb-creator               0.3.7
virtualenv                20.24.3
wadllib                   1.3.3
wcwidth                   0.2.5
webcolors                 1.13
webencodings              0.5.1
websocket-client          1.6.1
wheel                     0.34.2
widgetsnbextension        2.0.0
wrapt                     1.15.0
wxPython                  4.1.0
xarray                    2023.1.0
xdgenvpy                  2.3.5
xkit                      0.0.0
xlrd                      2.0.1
y-py                      0.6.0
yarl                      1.9.2
ypy-websocket             0.8.4
zipp                      3.8.1

**Comment by davidmeijer** [comment]
I have also noticed that using CellProfiler headless on our Linux Ubuntu server is very slow (about 30x slower than running it locally on my Macbook Pro). Curious to know if you ever found out what what slowing your analysis down?

**Comment by dana-mcc** [comment]
Unfortunately, I was unable to figure out what the issue was. I believe it was related to the install of CellProfiler and some inability to access files outside of test mode. It's not just that it was slow, but rather that there was no usage of the CPU at all that would indicate that a pipeline was running. I ended up borrowing a workstation with Windows installed.
